scene/gui/spin_box.cpp

```diff
@@ -68,6 +68,16 @@ void SpinBox::_text_submitted(const String &p_string) {
 	_value_changed(0);
 }
 
+void SpinBox::_text_changed(const String &p_string) {
+	int cursor_pos = line_edit->get_caret_column();
+
+	if (update_on_text_changed) {
+		_text_submitted(p_string);
+	}
+	// Line edit 'set_text' method resets the cursor position so we need to undo that.
+	line_edit->set_caret_column(cursor_pos);
+}
```

Comment:
This is going to be called on every text change, even if the user doesn't have `update_on_text_changed` enabled. Maybe it would be better to connect/disconnect the signal when this property changes.